src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -168,6 +189,34 @@ public void setDisableHttpsCertValidation(boolean disableHttpsCertValidation) {
     @CheckForNull
     public JnlpAgentEndpoint resolve() throws IOException {
         IOException firstError = null;
+
+        if (disableHttpEndpointCheck) {
+            if (jenkinsUrls.size() > 1) {
+                throw new IOException("Only one URL or Tunnel must be specified when HTTP Endpoint Check is disabled");
+            }
+
+            final String host;
+            final int port;
+            if (tunnel != null) {
+                String[] tokens = tunnel.split(":", 3);
+                if (tokens.length != 2)
+                    throw new IOException("Illegal tunneling parameter: " + tunnel);
+                host = tokens[0];
+                port = Integer.parseInt(tokens[1]);
+            } else if (!jenkinsUrls.isEmpty()) {
+               // URL like tcp://myjenkins:50000
+               URL url = new URL(jenkinsUrls.get(0));
```

Comment:
Use `URI` to parse, since actually using `tcp` will throw a `MalformedURLException` since that is not a real protocol. And recommend using `tcp` in documentation; it is very confusing to pass an `http` URL here, when there is no HTTP service there! For that matter, do we even need to support `-url tcp://myjenkins:50000` when `-tunnel myjenkins:50000` would do just as well? And do we even need the `-noHttpEndpoint` argument if `-tunnel` is being used?